interop-testing/src/main/java/io/grpc/testing/integration/TestServiceImpl.java

```diff
@@ -112,10 +124,45 @@ public void unaryCall(SimpleRequest req, StreamObserver<SimpleResponse> response
       return;
     }
 
+    try {
+      echoCallMetricsFromPayload(req.getOrcaPerRpcReport());
+    } catch (InvalidProtocolBufferException ex) {
+      responseObserver.onError(ex);
+    }
+    if (metricRecorder != null) {
+      try {
+        echoMetricsFromPayload(req.getOrcaOobReport());
+      } catch (InvalidProtocolBufferException ex) {
+        responseObserver.onError(ex);
+      }
+    }
     responseObserver.onNext(responseBuilder.build());
     responseObserver.onCompleted();
   }
 
+  private static void echoCallMetricsFromPayload(TestOrcaReport report)
+      throws InvalidProtocolBufferException {
```

Comment:
What line generates InvalidProtocolBufferException? Is this left over from the Any and not needed any more?